pkg/channelserver/channelserver.go

```diff
@@ -38,20 +38,22 @@ func GetURLAndInterval() (string, string) {
 
 }
 
-func run(driver, port, prefix string) chan error {
+func run() chan error {
```

Comment:
Is there a context we can pass in or create a context and use the cancel pattern here with it rather than making our own context pattern?